comp/rdnsquerier/impl/rdnsquerier.go

```diff
@@ -167,18 +168,67 @@ func (q *rdnsQuerierImpl) GetHostname(ipAddr []byte, updateHostnameSync func(str
 	}
 
 	if !netipAddr.IsPrivate() {
-		q.logger.Tracef("Reverse DNS Enrichment IP address %s is not in the private address space", netipAddr)
+		q.logger.Tracef("Reverse DNS Enrichment IP address %s is not in the private address space", ipAddr)
 		return nil
 	}
 	q.internalTelemetry.private.Inc()
 
 	err := q.cache.getHostname(netipAddr.String(), updateHostnameSync, updateHostnameAsync)
 	if err != nil {
 		q.logger.Debugf("Reverse DNS Enrichment cache.getHostname() for addr %s returned error: %v", netipAddr.String(), err)
-		return err
 	}
 
-	return nil
+	return err
+}
+
+// GetHostnameSync attempts to resolve the hostname for the given IP address synchronously.
+// If the IP address is invalid then an error is returned.
+// If the IP address is not in the private address space then it is ignored - no lookup is performed and nil error is returned.
+// If the IP address is in the private address space then the IP address will be resolved to a hostname.
+// The function accepts a timeout duration, which defaults to 2 seconds if not provided.
+func (q *rdnsQuerierImpl) GetHostnameSync(ctx context.Context, ipAddr string) (string, error) {
```

Comment:
Since the underlying implementation is async, shall we offer the possibly to pass multiple IPs to GetHostnameSync()?

For example, in Network Path case, for one traceroute, we might have many hops that requires reverse DNS, it would help to **concurrently** process rDNS for all hops (much faster processing time).

So, we would call `GetHostnameSync(ctx, []string[ip1, ip2, ip3])`, and request all 3 IPs async, wait for the all 3 results and return 3 results.

cc @ken-schneider

Reply:
That's a good plan, I'll go ahead and start on this